jobs/sql/incremental/newWatermarkQuery.sql

```diff
@@ -0,0 +1 @@
+select max(last_updated) from patient_last_update_date where partition_num = ${partitionNum}
```

Comment:
The new watermark is based on the max(last_updated date from patient_last_updated_date (from the dbevent_patient table).  The reason we do this is because this is ETL'd over first, and then the job that depends on it is run after.  In a real-time scenario, it's possible that the source would have later values than what is in the warehouse by the time we query it.  We want to enforce consistency where both the source and the target have the same knowledge about what the previous and future watermark dates will look like.

Reply:
to make sure I'm understanding, this patient_last_update_date table is in the target DB, correct, but just created from loading details from the db event source table?

Reply:
Correct.  We first ETL from dbevent_patient -> patient_last_update_date.  At this point, the data is fixed (whereby the dbevent_patient table is theoretically still changing while various jobs are executing, if we were in a real-time environment).  So all subsequent decisions about what data we already have, what the last watermark is, and what data has changed since that watermark, is all coming from the target db, which will give us consistency.